ui/build.gradle.kts

```diff
@@ -0,0 +1,253 @@
+
+import edu.wpi.first.wpilib.opencv.installer.Installer
+import org.codehaus.groovy.runtime.MethodClosure
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.Exec
+import org.gradle.api.tasks.testing.Test
+import org.gradle.api.tasks.testing.junit.JUnitOptions
+import org.gradle.script.lang.kotlin.*
+import org.gradle.util.GFileUtils
+
+buildscript {
+    repositories {
+        jcenter()
+        maven {
+            setUrl("https://github.com/WPIRoboticsProjects/opencv-maven/raw/mvn-repo")
+        }
+    }
+    dependencies {
+        classpath(dependencyNotation = "edu.wpi.first.wpilib.opencv:opencv-installer:2.0.0")
+        classpath(dependencyNotation = "de.dynamicfiles.projects.gradle.plugins:javafx-gradle-plugin:8.8.0")
+    }
+}
+
+plugins {
+    java
+    idea
+    jacoco
+    application
+}
+apply {
+    plugin("com.google.osdetector")
+    plugin("javafx-gradle-plugin")
+}
+
+val ideProviderConfiguration = configurations.maybeCreate("ideProvider")
+
+if (!(project.hasProperty("generation") || project.hasProperty("genonly"))) {
+    java {
+        sourceSets {
+            "test" {
+                java {
+                    exclude("**/ui/codegeneration")
+                    exclude("**/ui/codegeneration/**")
+                }
+            }
+        }
+    }
+}
+val coreProject = project(":core")
+dependencies {
+    compile(project(path = ":core", configuration = "shadow"))
```

Comment:
We could get rid of this extra level of complexity if we were to fully remove the deployer.